tests/e2e/vmgroupmisc_test.go

```diff
@@ -0,0 +1,269 @@
+	isDatastoreAccessible = admincli.IsDSAccessibleForVMgroup(s.config.EsxHost, vmGroupName, s.config.Datastores[1])
+	c.Assert(isDatastoreAccessible, Equals, false, Commentf("Datastore %s is accessible for vmgroup %s .", s.config.Datastores[1], vmGroupName))
+
+	// This volume create will fail because of trying to create volume on ds with no create privilege.
+	volumeName := inputparams.GetUniqueVolumeName(vmGroupName)
+	out, err = dockercli.CreateVolume(s.config.DockerHostNames[0], volumeName+"@"+s.config.Datastores[1])
+	c.Assert(err, Not(IsNil), Commentf(out))
+
+	// Check if volume was not created
+	isAvailable = verification.CheckVolumeAvailability(s.config.DockerHosts[0], volumeName+"@"+s.config.Datastores[1])
+	c.Assert(isAvailable, Equals, false, Commentf("Unexpected behavior: Volume %s is successfully created "+
+		" even though vmgroup [%s] does not have access rights for the datastore %s ", s.volumeNames[1], vmGroupName, s.config.Datastores[1]))
+
+	out, err = dockercli.DeleteVolume(s.config.DockerHosts[1], s.volumeNames[1])
+	c.Assert(err, IsNil, Commentf(out))
+
+	misc.LogTestEnd(c.TestName())
+}
+
+// Test steps:
+// 1. Create a vmgroup and associate a vm to it.
+// 2. Execute vmgroup ls command.
+// 3. Kill the vm that was added to the tenant.
+// 4. Again execute vmgroup ls command to verify command works fine.
+func (s *vgBasicSuite) TestVmgroupLsPostVMKill(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	powerState := esx.GetVMPowerState(s.config.DockerHostNames[0])
+	log.Printf("VM[%s]'s current power state is [%s]", s.config.DockerHostNames[0], properties.PowerOnState)
+	c.Assert(powerState, Equals, properties.PowerOnState, Commentf("VM [%s] should be powered on state", s.config.DockerHostNames[0]))
+
+	vmProcessID := esx.GetVMProcessID(s.config.EsxHost, s.config.DockerHostNames[0])
+	log.Printf("VM's process ID is: %s", vmProcessID)
+
+	// Kill the VM
+	isVMKilled := esx.KillVM(s.config.EsxHost, vmProcessID)
+	c.Assert(isVMKilled, Equals, true, Commentf("Unable to kill VM %s ...", s.config.DockerHostNames[0]))
+
+	isStatusChanged := esx.WaitForExpectedState(esx.GetVMPowerState, s.config.DockerHostNames[0], properties.PowerOffState)
+	c.Assert(isStatusChanged, Equals, true, Commentf("VM [%s] should be powered off state", s.config.DockerHostNames[0]))
+
+	// Check vmgroup ls
+	isVmgroupAvailable := admincli.IsVmgroupPresent(s.config.EsxHost, vmGroupName)
+	c.Assert(isVmgroupAvailable, Equals, true, Commentf("vmgroup ls command does not lists the vmgroup %s .", vmGroupName))
+
+	// Power-on the VM
+	esx.PowerOnVM(s.config.DockerHostNames[0])
+
+	isStatusChanged = esx.WaitForExpectedState(esx.GetVMPowerState, s.config.DockerHostNames[0], properties.PowerOnState)
+	c.Assert(isStatusChanged, Equals, true, Commentf("Failed to power-on the VM [%s]", s.config.DockerHostNames[0]))
+
+	// Verify vDVS successfully started
+	isVDVSRunning := verification.IsVDVSIsRunning(s.config.DockerHosts[0])
```

Comment:
it should be like vdvs is available meaning he plugin is able to accept requests vs. running. But not for this PR.